src/utils/thread_helper.cpp

```diff
@@ -968,10 +968,15 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
 
         wpanTopoFull->set_rloc16(rloc16);
 
-        otRouterInfo info;
+        {
+            otRouterInfo info;
 
-        VerifyOrExit(otThreadGetRouterInfo(mInstance, rloc16, &info) == OT_ERROR_NONE, error = OT_ERROR_INVALID_STATE);
-        wpanTopoFull->set_router_id(info.mRouterId);
+            // Store router info only when router is in correct status.
```

Comment:
The comment may not be accurate. It may return error when invalid arguments are given or the router cannot be found. I'd suggest not to describe the possible errors here.

Reply:
Done.